--- jbig2enc-0.23/Makefile.orig	2006-08-11 23:07:18.000000000 +0200
+++ jbig2enc-0.23/Makefile	2006-08-27 03:10:14.063467500 +0200
@@ -1,24 +1,25 @@
-CC=g++
+CC=gcc
+CXX=g++
 LEPTONICA=../leptonlib-1.53
 # For example, a fink MacOSX install:
 # EXTRA=-I/sw/include/ -I/sw/include/libpng -I/sw/include/libjpeg -L/sw/lib
-CFLAGS=-I${LEPTONICA}/src -Wall -I/usr/include -L/usr/lib -O3 ${EXTRA}
+CFLAGS=-I/usr/include/lept -Wall -I/usr/include ${EXTRA}
 
 jbig2: libjbig2enc.a jbig2.cc
-	$(CC) -o jbig2 jbig2.cc -L. -ljbig2enc -llept -L${LEPTONICA}/lib/nodebug $(CFLAGS) -lpng -ljpeg -ltiff -lm
+	$(CXX) -o jbig2 jbig2.cc -L. -ljbig2enc -llept $(CFLAGS) -lm
 
 libjbig2enc.a: jbig2enc.o jbig2arith.o jbig2sym.o
 	ar -rcv libjbig2enc.a jbig2enc.o jbig2arith.o jbig2sym.o
 
 jbig2enc.o: jbig2enc.cc jbig2arith.h jbig2sym.h jbig2structs.h jbig2segments.h
-	$(CC) -c jbig2enc.cc $(CFLAGS)
+	$(CXX) -c jbig2enc.cc $(CFLAGS)
 jbig2arith.o: jbig2arith.cc jbig2arith.h
-	$(CC) -c jbig2arith.cc $(CFLAGS)
+	$(CXX) -c jbig2arith.cc $(CFLAGS)
 jbig2sym.o: jbig2sym.cc jbig2arith.h
-	$(CC) -c jbig2sym.cc -DUSE_EXT $(CFLAGS)
+	$(CXX) -c jbig2sym.cc -DUSE_EXT $(CFLAGS)
 
 delta: delta.c
-	$(CC) -o delta delta.c $(CFLAGS) -llept -L${LEPTONICA}/lib/nodebug -lpng -ljpeg -ltiff -lm
+	$(CC) -o delta delta.c $(CFLAGS) -llept -lm
 
 clean:
 	rm -f *.o jbig2 libjbig2enc.a
